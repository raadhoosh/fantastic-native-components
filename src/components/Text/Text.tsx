import * as React from "react";
import Text from "./style.Text";
type Props = {
  theme?: any,
  children: any,
};
export default class App extends React.Component<Props> {
  props: any;
  render() {
    return (
      <Text {...this.props} >
        {this.props.children}
      </Text>
    );
  }
}