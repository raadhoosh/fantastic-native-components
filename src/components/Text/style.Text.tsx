import styled from "styled-components";
import { Text as RnText } from "react-native";
const Text = styled(RnText)`
color: ${(props: any) => props.theme.primary.main};
`;

export default Text;