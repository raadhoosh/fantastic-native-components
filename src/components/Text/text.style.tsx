import styled from "styled-components";
import { Text as RnText } from "react-native";
import { Theme } from '..';

interface IProps {
    primary?: boolean;
    secondary?: boolean;
    success?: boolean;
    info?: boolean;
    warning?: boolean;
    danger?: boolean;
    light?: boolean;
    dark?: boolean;
    bgColor?: string;
    color?: string;
    margin?: string;
    fontSize?: number
    theme?: Theme;
    fontWeight?: "normal" | "bold" | "100" | "200" | "300" | "400" | "500" | "600" | "700" | "800" | "900";
    textAlign?: "auto" | "left" | "right" | "center" | "justify",
}

function getColor(props: IProps) {

    let ForeColor = props.theme && props.theme.text.color ? props.theme.text.color : "#000";
    if (props.color) {
        ForeColor = props.color;
    } else {
        const color =
            (props.primary && "primary")
            || (props.secondary && "secondary")
            || (props.success && "success")
            || (props.info && "info")
            || (props.warning && "warning")
            || (props.danger && "danger");

        if (typeof (color) === "string" && props.theme) {
            ForeColor = props.theme[color].main;
        }

    }

    return ForeColor;
}

const ForeColor = (props: IProps) => getColor(props);

const Text = styled(RnText)`
color: ${ForeColor};
font-size:${(props: IProps) => props.fontSize ? props.fontSize : "14px"}; 
font-weight:${(props: IProps) => props.fontWeight ? props.fontWeight :
        (props.theme && props.theme.text.fontWeight ? props.theme.text.fontWeight : 'normal')}; 
text-align:${(props: IProps) => props.textAlign ? props.textAlign :
        (props.theme && props.theme.text.textAlign ? props.theme.text.textAlign : 'auto')}; 
width: auto       
`;

export default Text;