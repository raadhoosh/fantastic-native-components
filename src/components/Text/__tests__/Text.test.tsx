import Text from "../Text";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { View as RNView } from "react-native";

const createTestProps = (props: Object) => ({
  ...props,
});

describe("Text", () => {
  describe("rendering", () => {
    let wrapper: ShallowWrapper;
    let props: Object;
    beforeEach(() => {
      props = createTestProps({});
      wrapper = shallow(<Text {...props} />);
    });
    it("should render a <Text />", () => {
      const component = shallow(<Text />);
      expect(component).toMatchSnapshot();
    });
  });
});
