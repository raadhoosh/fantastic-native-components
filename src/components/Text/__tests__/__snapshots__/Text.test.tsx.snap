// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text rendering should render a <Text /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Text
    color="#f00"
    primary={true}
  >
    text
  </Text>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "text",
      "color": "#f00",
      "primary": true,
    },
    "ref": null,
    "rendered": "text",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
color: ",
          [Function],
          "; 
font-weight:",
          [Function],
          "; 
text-align:",
          [Function],
          "; 
font-size:",
          [Function],
          ";         
width: auto       
",
        ],
      },
      "displayName": "Styled(Component)",
      "propTypes": Object {
        "accessible": [Function],
        "adjustsFontSizeToFit": [Function],
        "allowFontScaling": [Function],
        "disabled": [Function],
        "ellipsizeMode": [Function],
        "minimumFontScale": [Function],
        "nativeID": [Function],
        "numberOfLines": [Function],
        "onLayout": [Function],
        "onLongPress": [Function],
        "onPress": [Function],
        "pressRetentionOffset": [Function],
        "selectable": [Function],
        "selectionColor": [Function],
        "style": [Function],
        "suppressHighlighting": [Function],
        "testID": [Function],
        "textBreakStrategy": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "text",
        "color": "#f00",
        "primary": true,
      },
      "ref": null,
      "rendered": "text",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
color: ",
            [Function],
            "; 
font-weight:",
            [Function],
            "; 
text-align:",
            [Function],
            "; 
font-size:",
            [Function],
            ";         
width: auto       
",
          ],
        },
        "displayName": "Styled(Component)",
        "propTypes": Object {
          "accessible": [Function],
          "adjustsFontSizeToFit": [Function],
          "allowFontScaling": [Function],
          "disabled": [Function],
          "ellipsizeMode": [Function],
          "minimumFontScale": [Function],
          "nativeID": [Function],
          "numberOfLines": [Function],
          "onLayout": [Function],
          "onLongPress": [Function],
          "onPress": [Function],
          "pressRetentionOffset": [Function],
          "selectable": [Function],
          "selectionColor": [Function],
          "style": [Function],
          "suppressHighlighting": [Function],
          "testID": [Function],
          "textBreakStrategy": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
